<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>452</width>
    <height>330</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>452</width>
    <height>330</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>452</width>
    <height>330</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Ngrok | GUI</string>
  </property>
  <property name="windowIcon">
   <iconset theme="application-default">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">import:url(:/np/material-blue.qss)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="5" column="2">
     <widget class="QPushButton" name="btn_stop">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>stop</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <spacer name="horizontalSpacer_4">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="1" rowspan="5" colspan="2">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="minimumSize">
       <size>
        <width>440</width>
        <height>60</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLineEdit" name="ngurl">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">color: rgb(255, 255, 255);</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="frame">
          <bool>true</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="clearButtonEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="ng_copy">
         <property name="text">
          <string>copy</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QPushButton" name="btn_start">
      <property name="text">
       <string>start</string>
      </property>
     </widget>
    </item>
    <item row="7" column="1" colspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string/>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QTabWidget" name="tabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="setting">
          <attribute name="title">
           <string>setting</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Token : </string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="ng_token">
               <property name="maxLength">
                <number>60</number>
               </property>
               <property name="placeholderText">
                <string>eg: DZIXnhJt2HNWLXyQ5PM_qCaBq0W2sNFcCa0rfTZd</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Port :    </string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="ng_port">
               <property name="text">
                <string>80</string>
               </property>
               <property name="maxLength">
                <number>5</number>
               </property>
               <property name="placeholderText">
                <string>eg: 8080</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QRadioButton" name="ng_port_tcp">
               <property name="text">
                <string>TCP</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QRadioButton" name="ng_port_http">
               <property name="text">
                <string>http</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Subdomain : </string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="ng_subdomain">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="frame">
                <bool>true</bool>
               </property>
               <property name="readOnly">
                <bool>false</bool>
               </property>
               <property name="placeholderText">
                <string>eg: john (pro only)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="ng_subdomain_enable">
               <property name="text">
                <string>Enable</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="log">
          <attribute name="title">
           <string>log</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="0">
            <widget class="QTextBrowser" name="ng_log">
             <property name="textInteractionFlags">
              <set>Qt::TextSelectableByMouse</set>
             </property>
             <property name="openLinks">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="5" column="3">
     <spacer name="horizontalSpacer_5">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionRemove_Token">
   <property name="icon">
    <iconset theme="remove">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Remove Token</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionExport_Token">
   <property name="icon">
    <iconset theme="blueman-send-file">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Export Token</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionQuit_2">
   <property name="icon">
    <iconset theme="exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout_ngrok">
   <property name="icon">
    <iconset theme="info">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>About ngrok</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="icon">
    <iconset theme="qt">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionGitHub">
   <property name="icon">
    <iconset theme="github">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>GitHub</string>
   </property>
  </action>
  <action name="actionTelegram">
   <property name="icon">
    <iconset theme="telegram">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Telegram</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="info">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>ng_subdomain_enable</sender>
   <signal>toggled(bool)</signal>
   <receiver>ng_subdomain</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>426</x>
     <y>298</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ng_port_tcp</sender>
   <signal>toggled(bool)</signal>
   <receiver>ng_subdomain_enable</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel">
     <x>391</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ng_port_tcp</sender>
   <signal>clicked(bool)</signal>
   <receiver>ng_subdomain</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
